test/unit/mdl-textfield/foundation.test.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import test from 'tape';
+import td from 'testdouble';
+
+import {verifyDefaultAdapter} from '../helpers/foundation';
+import {setupFoundationTest} from '../helpers/setup';
+import MDLTextfieldFoundation from '../../../packages/mdl-textfield/foundation';
+
+const {cssClasses} = MDLTextfieldFoundation;
+
+test('exports strings', t => {
+  t.true('strings' in MDLTextfieldFoundation);
+  t.end();
+});
+
+test('exports cssClasses', t => {
+  t.true('cssClasses' in MDLTextfieldFoundation);
+  t.end();
+});
+
+test('defaultAdapter returns a complete adapter implementation', t => {
+  verifyDefaultAdapter(MDLTextfieldFoundation, [
+    'addClass', 'removeClass', 'addClassToLabel', 'removeClassFromLabel',
+    'addClassToHelptext', 'removeClassFromHelptext', 'helptextHasClass',
+    'registerInputFocusHandler', 'deregisterInputFocusHandler',
+    'registerInputBlurHandler', 'deregisterInputBlurHandler',
+    'setHelptextAttr', 'rmHelptextAttr', 'getNativeInput'
+  ], t);
+  t.end();
+});
+
+const setupTest = () => setupFoundationTest(MDLTextfieldFoundation);
+
+test('#constructor sets disabled to false', t => {
+  const {foundation} = setupTest();
+  t.false(foundation.isDisabled());
+  t.end();
```

Comment:
Tape is minimal and async by default, so this is how it handles signalling that a test is over.

```js
test('#get() fetches API data', t => {
  const response = {foo: 'bar'};
  fakeAPI.whenGet('/data').thenRespond(JSON.stringify(response));

  fetcher.getData().then(data => {
    t.deepEqual(data, response);
    t.end();
  });
});
```

There's also [assertion planning](https://github.com/substack/tape#tplann), but our tape-eslint plugin doesn't support it and always having to change the number of assertions in a test gets unwieldy.